### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by DrRey**

### CODE REVIEW

File: buildSrc/src/main/kotlin/Config.kt [hunk 1]
```diff
@@ -1,33 +1,33 @@
 object Config {
```
Comment (petersamokhin) [inline buildSrc/src/main/kotlin/Config.kt]: pls remove the whole buildSrc and let's migrate to gradle versions catalog :)

File: build.gradle.kts [hunk 2]
```diff
@@ -17,17 +16,12 @@ allprojects {
     tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile::class).all {
         JavaVersion.VERSION_1_8.toString().also {
```
Comment (petersamokhin) [inline build.gradle.kts]: since kotlin 1.5 it is no longer required and can be removed

File: auth/build.gradle.kts [hunk 3]
```diff
@@ -29,13 +29,12 @@ android {
...
     sourceSets {
-        val main by getting {}
-        val test by getting {}
-        val androidTest by getting {}
+        val main by getting
+        val test by getting
+        val androidTest by getting
         main.java.srcDirs("src/main/kotlin")
         test.java.srcDirs("src/test/kotlin")
         androidTest.java.srcDirs("src/androidTest/kotlin")
```
Comment (petersamokhin) [inline auth/build.gradle.kts]: seems redundant probably, could you pls check?

File: auth/build.gradle.kts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import com.android.build.gradle.LibraryExtension
```
Comment (petersamokhin) [inline auth/build.gradle.kts]: please inline it, imports here look weird to me 😄

File: auth/build.gradle.kts [hunk 5]
```diff
@@ -6,17 +8,15 @@ plugins {
...
     defaultConfig {
-        minSdkVersion(Config.Android.minSdk)
-        targetSdkVersion(Config.Android.compileSdk)
-        buildToolsVersion(Config.Android.buildToolsVersion)
+        minSdk = Config.Android.minSdk
+        targetSdk = Config.Android.compileSdk
+        buildToolsVersion = Config.Android.buildToolsVersion
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
 
         // applicationId = "${project.property("GROUP").toString()}.auth"
```
Comment (petersamokhin) [inline auth/build.gradle.kts]: could be removed

File: buildSrc/src/main/kotlin/Config.kt [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
     object Versions {
         object Kotlin {
-            const val kotlin = "1.3.72"
+            const val kotlin = "1.7.22"
         }
         object Plugin {
-            const val androidGradle = "3.6.1"
-            const val publish = "0.11.1"
+            const val androidGradle = "7.3.1"
+            const val publish = "0.22.0"
```
Comment (petersamokhin) [inline buildSrc/src/main/kotlin/Config.kt]: I see you migrated the logic to maven-publish, this one can be removed from dependencies then

**Review by petersamokhin — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/Config.kt (hunk 1 above)
Comment (DrRey) [reply buildSrc/src/main/kotlin/Config.kt]: done

File: buildSrc/src/main/kotlin/Config.kt (hunk 6 above)
Comment (DrRey) [reply buildSrc/src/main/kotlin/Config.kt]: It should still use gradle-maven-publish-plugin, with an updated config syntax.
The POM probably needs to be updated with some details.
https://vanniktech.github.io/gradle-maven-publish-plugin/other/

File: settings.gradle.kts [hunk 7]
```diff
@@ -7,4 +7,18 @@ include(
...
+)
+
+pluginManagement {
+    repositories {
+        google()
+        gradlePluginPortal()
+    }
+    resolutionStrategy {
+        eachPlugin {
+            if(requested.id.namespace == "com.android") {
```
Comment (petersamokhin) [inline settings.gradle.kts]:
```suggestion
            if (requested.id.namespace == "com.android") {
```

**Review by petersamokhin — APPROVED** [review]

**Pull request merged**